esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -1116,3 +1116,11 @@ pub fn get_wakeup_cause() -> SleepSource {
 
     SleepSource::Undefined
 }
+
+// libphy.a can pull this in on some chips, we provide it here in the hal
+// so that either ieee or esp-wifi's get it for free without duplicating in both
+#[no_mangle]
+extern "C" fn rtc_clk_xtal_freq_get() -> i32 {
```

Comment:
Nice! I wanted to do this (or something similar) for quite some time but always forgot about it

Reply:
Weeee, finally, we have real frequencies :D